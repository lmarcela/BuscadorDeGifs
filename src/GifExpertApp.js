import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {
    const [categories, setCategories] = useState( (defaultCategories.length>0) ? defaultCategories: ["oso"]);
      
    //setCategories([...categories, "Gatos"]);
    //setCategories( c => [...c, "Gatos"]);
    
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            
            <div className="w3-container">
                {
                    categories.map(category => (
                        <GifGrid key={category} category={category} />
                    ))
                }
            </div>
        </>
    )
}

  export default GifExpertApp;