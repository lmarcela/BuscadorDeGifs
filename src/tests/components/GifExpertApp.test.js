import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifExpertApp} from '../../GifExpertApp';

describe('Pruebas en <GifExpertApp />',()=>{      

    test('debe mostrar <GifExpertApp /> correctamente', () => {     
        const wrapper=shallow(<GifExpertApp />); 
        expect(wrapper).toMatchSnapshot();
    })    

    test('debe de mostrar una lista de categorias', () => {     
        const categorias = ['ferdinand','puppy'];
        const wrapper=shallow(<GifExpertApp defaultCategories={categorias}/>); 

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })

});


