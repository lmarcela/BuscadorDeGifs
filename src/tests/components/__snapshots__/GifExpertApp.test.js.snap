// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="w3-container"
  >
    <GifGrid
      category="ferdinand"
      key="ferdinand"
    />
    <GifGrid
      category="puppy"
      key="puppy"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="w3-container"
  >
    <GifGrid
      category="oso"
      key="oso"
    />
  </div>
</Fragment>
`;
